### PR TITLE
[fix] Update bincode

### PR DESCRIPTION
**Pull request by warittornc**
## Implementation details
- Do 1
- Do 2

## Please ensure the following requirements are met before submitting a pull request:

- [ ] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the GitHub PR explorer)

### CODE REVIEW

File: bothan-api/server-cli/src/commands/start.rs [hunk 1]
```diff
@@ -93,6 +93,7 @@ async fn start_server(
     let ipfs_client = init_ipfs_client(&app_config).await?;
     let crypto_server = Arc::new(init_crypto_server(&app_config, store, ipfs_client).await?);
 
+    println!("Server started");
```
Comment (sorawit) [inline bothan-api/server-cli/src/commands/start.rs:96]: remove this. or change to tracing?

File: bothan-core/src/registry/post_processor.rs [hunk 2]
```diff
@@ -22,7 +23,7 @@ pub trait PostProcess<T> {
 }
 
 /// The PostProcess enum represents the different types of post-processors that can be used.
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, Encode, Decode)]
```
Comment (sorawit) [inline bothan-core/src/registry/post_processor.rs:26]: if we add more enum variants will it be backward-compatible?

**Review by sorawit — DISMISSED** [review]

File: bothan-api/server/src/api/crypto.rs [hunk 3]
```diff
@@ -53,23 +53,23 @@ impl SignalService for CryptoQueryServer {
...
-                error!("Failed to parse registry");
+                error!("failed to parse registry");
                 Err(Status::invalid_argument("Registry is invalid"))
             }
             Err(SetRegistryError::InvalidHash) => {
-                error!("Invalid IPFS hash");
+                error!("invalid IPFS hash");
                 Err(Status::invalid_argument("Invalid IPFS hash"))
             }
             Err(SetRegistryError::FailedToSetRegistry(_)) => {
```
Comment (colmazia) [inline bothan-api/server/src/api/crypto.rs:75]: can't we include the internal error?

File: bothan-core/src/store.rs [hunk 4]
```diff
@@ -55,15 +56,16 @@ impl SharedStore {
     pub async fn restore(&mut self) -> Result<(), Error> {
         let mut inner = self.inner.write().await;
 
-        let registry = inner
+        let invalid_registry: Option<Registry> = inner
```
Comment (colmazia) [inline bothan-core/src/store.rs:59]: consider raw_registry, pending_registry, pre_validate_registry

File: bothan-core/src/registry/post_processor.rs (hunk 2 above)
Comment (warittornc) [reply bothan-core/src/registry/post_processor.rs:26]: yes, bincode encodes enum as u32 so adding more shouldn't be an issue

File: bothan-core/src/store.rs [hunk 5]
```diff
@@ -55,15 +56,16 @@ impl SharedStore {
...
             .db
             .get(Key::Registry.to_prefixed_bytes())?
-            .map(|b| bincode::deserialize(b.as_slice()))
-            .transpose()?;
+            .map(|b| decode_from_slice(b.as_slice(), config::standard()))
+            .transpose()?
+            .map(|(r, _)| r);
 
-        if let Some(registry) = registry {
+        if let Some(invalid_reg) = unvalidated_registry {
```
Comment (RogerKSI) [inline bothan-core/src/store.rs:66]: invalid?

**Review by RogerKSI — APPROVED** [review]

**Review by colmazia — APPROVED** [review]

**Pull request merged**